web/src/main/java/com/bazaarvoice/emodb/web/scanner/writer/KafkaScanWriter.java

```diff
@@ -0,0 +1,82 @@
+package com.bazaarvoice.emodb.web.scanner.writer;
+
+import com.bazaarvoice.emodb.kafka.KafkaCluster;
+import com.codahale.metrics.MetricRegistry;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+import com.google.inject.assistedinject.Assisted;
+import java.io.IOException;
+import java.net.URI;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.kafka.clients.producer.Producer;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class KafkaScanWriter implements ScanWriter {
+
+    private final Producer<String, JsonNode> _producer;
+    private final ObjectMapper _mapper;
+    private final Clock _clock;
+    private final MetricRegistry _metricRegistry;
+    private final String _topicName;
+    private final Map<TransferKey, KafkaScanDestinationWriter> _writers;
+
+    @Inject
+    public KafkaScanWriter(KafkaCluster kafkaCluster, ObjectMapper objectMapper, Clock clock,
+                           MetricRegistry metricRegistry, @Assisted URI baseUri) {
+
+        requireNonNull(kafkaCluster, "kafkaCluster");
+
+        _producer = kafkaCluster.producer();
+        _mapper = requireNonNull(objectMapper, "objectMapper");
+        _clock = requireNonNull(clock, "clock");
+        _topicName  = requireNonNull(baseUri.getHost(), "topicName");
+        _metricRegistry = requireNonNull(metricRegistry, "metricRegistry");
+        checkArgument(!_topicName.isEmpty());
+        _writers = new HashMap<>();
```

Comment:
could be assigned in place

Reply:
I prefer to do all non static assignment in the constructor.

Reply:
that makes sense, but field is final and doesn't depend on constructor parameters.

Reply:
agreed. If you feel super strongly about this, lmk and I will change it.